drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1256 @@
+ *
+ * @return                  CRC length in [bytes]
+ */
+static inline uint8_t nrf24l01p_etoval_crco(nrf24l01p_crc_t crc_len)
+{
+    if (crc_len <= NRF24L01P_CRC_0BYTE) {
+        return 0;
+    }
+    if (crc_len == NRF24L01P_CRC_1BYTE) {
+        return 1;
+    }
+    return 2;
+}
+
+/**
+ * @brief   Convert CRC length in [bytes] to @ref nrf24l01p_crc_t
+ *
+ * @param[in] crc_len       CRC length in [bytes]
+ *
+ * @return                  Corresponding enum
+ */
+static inline nrf24l01p_crc_t nrf24l01p_valtoe_crco(uint8_t crc_len)
+{
+    if (!crc_len) {
+        return NRF24L01P_CRC_0BYTE;
+    }
+    if (crc_len == 1) {
+        return NRF24L01P_CRC_1BYTE;
+    }
+    return NRF24L01P_CRC_2BYTE;
+}
+
+/**
+ * @brief   Convert @ref nrf24l01p_tx_power_t to actual Tx power
+ *
+ * @param[in] power         RF power enum
+ *
+ * @return                  RF power in [dbm]
+ */
+static inline int8_t nrf24l01p_etoval_rfpwr(nrf24l01p_tx_power_t power)
+{
+    if (power <= NRF24L01P_TX_POWER_MINUS_18DBM) {
+        return -18;
+    }
+    if (power == NRF24L01P_TX_POWER_MINUS_12DBM) {
+        return -12;
+    }
+    if (power == NRF24L01P_TX_POWER_MINUS_6DBM) {
+        return -6;
+    }
+    return 0;
+}
+
+/**
+ * @brief   Convert RF power in [dbm] to @ref nrf24l01p_tx_power_t
+ *
+ * @param[in] power         RF power in [dbm]
+ *
+ * @return                  Corresponding enum
+ */
+static inline nrf24l01p_tx_power_t nrf24l01p_valtoe_rfpwr(int16_t power)
+{
+    if (power <= -18) {
+        return NRF24L01P_TX_POWER_MINUS_18DBM;
+    }
+    if (power <= -12) {
+        return NRF24L01P_TX_POWER_MINUS_12DBM;
+    }
+    if (power <= -6) {
+        return NRF24L01P_TX_POWER_MINUS_6DBM;
+    }
+    return NRF24L01P_TX_POWER_0DBM;
+}
+
+/**
+ * @brief   Convert @ref nrf24l01p_rfdr_t to actual air data rate
+ *
+ * @param[in] data_rate     Air data rate enum
+ *
+ * @return                  Air data rate in [kbit/s]
+ */
+static inline uint16_t nrf24l01p_etoval_rfdr(nrf24l01p_rfdr_t data_rate)
+{
+    if (data_rate <= NRF24L01P_RF_DR_1MBPS) {
+        return 1000;
+    }
+    if (data_rate == NRF24L01P_RF_DR_250KBPS) {
+        return 250;
+    }
+    return 2000;
+}
+
+/**
+ * @brief   Convert Air data rate in [kbit/s] to @ref nrf24l01p_rfdr_t
+ *
+ * @param[in] data_rate     Air data rate in [kbit/s]
+ *
+ * @return                  Corresponding enum
+ */
+static inline nrf24l01p_rfdr_t nrf24l01p_valtoe_rfdr(uint16_t data_rate)
+{
+    if (data_rate <= 250) {
+        return NRF24L01P_RF_DR_250KBPS;
+    }
+    if (data_rate <= 1000) {
+        return NRF24L01P_RF_DR_1MBPS;
+    }
+    return NRF24L01P_RF_DR_2MBPS;
+}
+
+/**
+ * @brief   Wrapper around @see nrf24l01p_set_mtu to set the
+ *          payload width of pipe 0
+ *
+ * @param[in] dev       NRf24L01P device handle
+ * @param[in] width     paylaod width
+ *
+ * @return              @see nrf24l01p_set_mtu
+ */
+static inline int nrf24l01p_set_mtu_p0(nrf24l01p_t *dev, uint8_t width)
```

Comment:
Yes

Reply:
So far, I have not moved `nrf24l01p_set_mtu()` and `nrf24l01p_get_mtu()` to the advanced API because`nrf24l01p_get_mtu()` is needed by `netdev::nrf24l01p_netdev::nrf24l01p_get()` and an advanced API pseudomodule would no longer be optional.

I could only move `nrf24l01p_set_mtu()` to the advanced API, but splitting set/get functions doesn´t feel kind of right, to me